conan/internal/graph/graph_binaries.py

```diff
@@ -450,6 +450,8 @@ def _evaluate_single(n):
             # Evaluate the possible nodes with repeated "prefs" that haven't been evaluated
             for pref, pref_nodes in nodes.items():
                 for n in pref_nodes[1:]:
+                    # Ensure that same prefs means same conanfile
+                    n.conanfile = pref_nodes[0].conanfile
```

Comment:
This alternative approach would ensure that nodes with the same pref actually share the conanfile instance, which should be identical in all branches unless I'm severly mistaken

Reply:
But this is making the tests fail in ways I hadn't anticipate